judge/models/problem.py

```diff
@@ -449,6 +462,31 @@ def save(self, *args, **kwargs):
 
     save.alters_data = True
 
+    def solved_by(self, user):
+        # Return true if a full AC submission to the problem from the user exists.
+        return self.submission_set.filter(user=user.profile, result='AC', points=F('problem__points')).exists()
```

Comment:
Can we change this to `>= problem_points` (yes, there are such problems, and yes, they are silly).

Reply:
done (this also fixed an existing bug with `source_visibility == SubmissionSourceAccess.SOLVED`)